src/utilities/memory.c

```diff
@@ -495,12 +501,14 @@ hypre_Free_core(void *ptr, hypre_MemoryLocation location)
       return;
    }
 
+#if defined(HYPRE_USING_CUDA) || defined(HYPRE_USING_HIP)
```

Comment:
`hypre_GerPointerLocation()` is also available with sycl, so you could add `|| defined(HYPRE_USING_SYCL)` here.

Reply:
Fixed. Thanks!